sdk/armcore/connection.go

```diff
@@ -83,16 +91,13 @@ func NewConnection(endpoint string, cred azcore.TokenCredential, options *Connec
 		}
 		policies = append(policies, NewRPRegistrationPolicy(endpoint, cred, &regRPOpts))
 	}
-	policies = append(policies, azcore.NewRetryPolicy(&options.Retry),
+	policies = append(policies, options.PerCallPolicies...)
+	policies = append(policies, azcore.NewRetryPolicy(&options.Retry))
+	policies = append(policies, options.PerRetryPolicies...)
```

Comment:
I placed this before the authentication policy to prevent tampering with the credential.  Any objections?